### ISSUE TITLE
EXC_BAD_ACCESS within terrain_renderer

### ISSUE THREAD

**Issue by tomascern** [mapbox/mapbox-maps-ios#1177]
## Environment

- Xcode version: 13.2.1
- iOS version: 15.3.1
- Devices affected: All
- Maps SDK Version: 10.3.0

## Observed behavior and steps to reproduce

After setting terrain on in offline mode, it shows up but eventually after some map panning and zooming the app crashes with the EXC_BAD_ACCESS. I am using OfflineRegionManager for offline data. See attached stacktrace.

```
* thread #1, queue = 'com.apple.main-thread', stop reason = EXC_BAD_ACCESS (code=1, address=0x20)
  * frame #0: 0x000000010b4203f8 MapboxCoreMaps`::upload() [inlined] getLoadedAreaMinimum at terrain_renderer.cpp:732:34 [opt]
    frame #1: 0x000000010b4202f8 MapboxCoreMaps`::upload() [inlined] updateEmptyDEMTexture at terrain_renderer.cpp:743:46 [opt]
    frame #2: 0x000000010b4202f8 MapboxCoreMaps`::upload() at terrain_renderer.cpp:452:5 [opt]
    frame #3: 0x000000010b3ee56c MapboxCoreMaps`::render() at renderer_impl.cpp:160:22 [opt]
    frame #4: 0x000000010b3e8e84 MapboxCoreMaps`::render() at renderer.cpp:51:15 [opt]
    frame #5: 0x000000010b1bcdc4 MapboxCoreMaps`::render() [inlined] render at map_renderer_impl.cpp:64:16 [opt]
    frame #6: 0x000000010b1bcdb8 MapboxCoreMaps`::render() [inlined] render at map_renderer.cpp:22:12 [opt]
    frame #7: 0x000000010b1bccf0 MapboxCoreMaps`::render() at map_impl.cpp:105:23 [opt]
    frame #8: 0x000000010b1bb2ec MapboxCoreMaps`-[MBMMetalViewDelegate drawInMTKView:] at view_delegate.mm:9:28 [opt]
    frame #9: 0x00000001bdbdd694 MetalKit`-[MTKView draw] + 184
    frame #10: 0x00000001847b00d0 QuartzCore`CA::Layer::layout_and_display_if_needed(CA::Transaction*) + 428
    frame #11: 0x00000001847c45dc QuartzCore`CA::Context::commit_transaction(CA::Transaction*, double, double*) + 524
    frame #12: 0x00000001847cced0 QuartzCore`CA::Transaction::commit() + 680
    frame #13: 0x00000001847af70c QuartzCore`CA::Transaction::flush_as_runloop_observer(bool) + 88
    frame #14: 0x0000000183410208 UIKitCore`_UIApplicationFlushCATransaction + 76
    frame #15: 0x00000001836aab44 UIKitCore`_UIUpdateSequenceRun + 84
    frame #16: 0x0000000183d2784c UIKitCore`schedulerStepScheduledMainSection + 144
    frame #17: 0x0000000183d26e3c UIKitCore`runloopSourceCallback + 60
    frame #18: 0x0000000180adf0d0 CoreFoundation`__CFRUNLOOP_IS_CALLING_OUT_TO_A_SOURCE0_PERFORM_FUNCTION__ + 28
    frame #19: 0x0000000180aefd90 CoreFoundation`__CFRunLoopDoSource0 + 208
    frame #20: 0x0000000180a2a104 CoreFoundation`__CFRunLoopDoSources0 + 376
    frame #21: 0x0000000180a2f8a4 CoreFoundation`__CFRunLoopRun + 820
    frame #22: 0x0000000180a43468 CoreFoundation`CFRunLoopRunSpecific + 600
    frame #23: 0x000000019c5e738c GraphicsServices`GSEventRunModal + 164
    frame #24: 0x00000001833e65d0 UIKitCore`-[UIApplication _run] + 1100
    frame #25: 0x0000000183164f74 UIKitCore`UIApplicationMain + 364
    frame #26: 0x0000000198328fa0 libswiftUIKit.dylib`UIKit.UIApplicationMain(Swift.Int32, Swift.Optional<Swift.UnsafeMutablePointer<Swift.UnsafeMutablePointer<Swift.Int8>>>, Swift.Optional<Swift.String>, Swift.Optional<Swift.String>) -> Swift.Int32 + 104
    frame #27: 0x00000001031066f4 app`static UIApplicationDelegate.main() at <compiler-generated>:0
    frame #28: 0x000000010310667c app`static AppDelegate.$main(self=luftLocSim.AppDelegate) at AppDelegate.swift:11:1
    frame #29: 0x0000000103106774 app`main at <compiler-generated>:0
    frame #30: 0x0000000104159aa4 dyld`start + 520
```

## Expected behavior

Terrain shows up and works without crash even in offline mode.

**Comment by ZiZasaurus** [comment]
@tomascern, improvements to offline 3D terrain will be addressed in v10.4 of the Maps SDK.

**Comment by ZiZasaurus** [comment]
Closing this ticket, but please feel free to reopen if you have additional questions.
